### PR TITLE
Replace gulp with modern tools or npm scripts

### PR DESCRIPTION
**Pull request by allmywant**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,27 +1,26 @@
 {
   "name": "mojito-js-delivery",
-  "version": "2.4.1",
+  "version": "2.4.2",
```
Comment (kingo55) [inline package.json]: Is it possible to bump this to 2.5.0? This is quite a chunky enhancement and has some big changes to the CLI and Pipelines.

**Review by kingo55 — APPROVED** [review]
This looks great @allmywant - can't wait to merge this in...

File: package.json (hunk 1 above)
Comment (allmywant) [reply package.json]: @kingo55 
Done.

File: scripts/cli-build.js [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    if ('css' in testObject) {
+        if (testObject.css != '') {
+            testObject.css = '{{' + testObject.css + '}}';
+        }
+        else {
+            delete testObject.css;
+        }
+    }
+
+    // trigger. TODO: throw error if trigger is undefined?
```
Comment (dapperdrop) [inline scripts/cli-build.js:113]: I think throwing an error here is a good idea. Tests without triggers won't work.

**Review by kingo55 — APPROVED** [review]

File: scripts/cli-build.js (hunk 2 above)
Comment (kingo55) [reply scripts/cli-build.js:113]: Unless you activate them remotely. I'm pretty sure this is possible (it's just not a typical way to activate experiments).

Comment (allmywant) [reply scripts/cli-build.js:113]: @dapperdrop 
A test without trigger can be run by calling test.activate() manually in other tests, let's keep it as is for now?

Comment (dapperdrop) [reply scripts/cli-build.js:113]: Good points @kingo55 @allmywant - all good to keep as is for now.

**Review by dapperdrop — APPROVED** [review]

**Comment by kingo55** [comment]
Glorious... thanks for this epic PR @allmywant

**Pull request merged**